snap/info.go

```diff
@@ -269,24 +269,25 @@ func (app *AppInfo) WrapperPath() string {
 }
 
 func (app *AppInfo) launcherCommand(command string) string {
-	securityTag := app.SecurityTag()
-	return fmt.Sprintf("/usr/bin/ubuntu-core-launcher %s %s %s", securityTag, securityTag, filepath.Join(app.Snap.MountDir(), command))
-
+	if app.Name == app.Snap.Name() {
+		return fmt.Sprintf("/usr/bin/snap run %s %s", command, filepath.Base(app.Name))
+	}
+	return fmt.Sprintf("/usr/bin/snap run %s %s.%s", command, app.Snap.Name(), filepath.Base(app.Name))
```

Comment:
Can we please do something like this above:
```
if command != '" {
        command = " " + command
}
```
and then tweak the templates as `"... run%s %s.%s"`?

This will ensure we're not getting these double whitespaces between run and the app name.